odxtools/endofpdufield.py

```diff
@@ -67,32 +33,13 @@ def from_et(et_element: ElementTree.Element,
         else:
             max_number_of_items = None
 
-        is_visible_raw = odxstr_to_bool(et_element.get("IS-VISIBLE"))
         eopf = EndOfPduField(
-            sdgs=sdgs,
-            structure_ref=structure_ref,
-            structure_snref=structure_snref,
-            env_data_desc_ref=env_data_desc_ref,
-            env_data_desc_snref=env_data_desc_snref,
             min_number_of_items=min_number_of_items,
             max_number_of_items=max_number_of_items,
-            is_visible_raw=is_visible_raw,
             **kwargs)
 
         return eopf
 
-    @property
-    def structure(self) -> "BasicStructure":
-        """may be a Structure or a env-data-desc"""
-        return self._structure
```

Comment:
@andlaus it seems env-data-desc was broken in existing code of endofpdufield

Reply:
true: We don't test it in the unittests and I also do not have any files which use ENV-DATA-DESC in END-OF-PDUFIELDs. Let's try to do it right this time (at least in principle); Without exercising this code, I have some doubts if all details will be correct, but let's burn that bridge when we come to it...

Reply:
I updated the code, it still does not support ENV-DATA-DESC, but it least it reports it correctly once detected